addons/website_slides/models/slide_slide.py

```diff
@@ -230,30 +230,49 @@ def _compute_total(self):
             record.total_views = record.slide_views + record.public_views
 
     @api.depends('slide_partner_ids.vote')
-    @api.depends_context('uid')
-    def _compute_user_info(self):
-        default_stats = {'likes': 0, 'dislikes': 0, 'user_vote': False}
+    def _compute_like_info(self):
+        default_stats = {'likes': 0, 'dislikes': 0}
 
         if not self.ids:
             self.update(default_stats)
             return
 
-        slide_data = dict.fromkeys(self.ids, default_stats)
+        slide_data = dict(
+            (record_id, dict(default_stats))
+            for record_id in self.ids
+        )
         slide_partners = self.env['slide.slide.partner'].sudo().search([
-            ('slide_id', 'in', self.ids)
+            ('slide_id', 'in', self.ids),
         ])
         for slide_partner in slide_partners:
             if slide_partner.vote == 1:
                 slide_data[slide_partner.slide_id.id]['likes'] += 1
-                if slide_partner.partner_id == self.env.user.partner_id:
-                    slide_data[slide_partner.slide_id.id]['user_vote'] = 1
             elif slide_partner.vote == -1:
                 slide_data[slide_partner.slide_id.id]['dislikes'] += 1
-                if slide_partner.partner_id == self.env.user.partner_id:
-                    slide_data[slide_partner.slide_id.id]['user_vote'] = -1
+
         for slide in self:
             slide.update(slide_data[slide.id])
 
+    @api.depends('slide_partner_ids.vote')
+    @api.depends_context('uid')
+    def _compute_user_info(self):
+        self.user_vote = 0
+
+        if not self.ids:
+            return
+
+        slide_data = dict.fromkeys(self.ids, 0)
+        slide_partners = self.env['slide.slide.partner'].sudo().search([
+            ('slide_id', 'in', self.ids),
+            ('partner_id', '=', self.env.user.partner_id.id),
+        ])
+        for slide_partner in slide_partners:
+            if slide_partner.vote in (1, -1):
+                slide_data[slide_partner.slide_id.id] = slide_partner.vote
+
+        for slide in self:
+            slide.user_vote = slide_data.get(slide.id, 0)
```

Comment:
Since you use a fromkeys(), not sure the default value is necessary.
Or you could just initialize "slide_data" to a regular dict.

(Or don't change anything, I mean it's no big deal).

Reply:
I hesitated and kept the .get, just in case. Maybe in stable we can keep the default + get and cleanup a bit in master.